GUI/Controls/TreeViewWithSearchResults.cs

```diff
@@ -0,0 +1,250 @@
+                else if (e.Button == MouseButtons.Left)
+                {
+                    var node = info.Item.Tag as TreeNode;
+                    if (node.Tag is TreeViewFolder)
+                    {
+                        node.EnsureVisible();
+                        node.TreeView.SelectedNode = node;
+                    }
+                }
+            }
+            else
+            {
+                mainListView.SelectedItems.Clear();
+            }
+        }
+
+        /// <summary>
+        /// If the user double clicks (with left mouse button) on a ListViewItem, send up an event to subscribers that such an action has occurred. Also send up
+        /// whatever object is represented by the ListViewItem.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void MainListView_MouseDoubleClick(object sender, MouseEventArgs e)
+        {
+            if (e.Button == MouseButtons.Left)
+            {
+                ListViewHitTestInfo info = mainListView.HitTest(e.X, e.Y);
+
+                if (info.Item != null)
+                {
+                    // if user left double clicks a folder, open its contents and display in list view
+                    var node = info.Item.Tag as TreeNode;
+                    if (node.Tag is TreeViewFolder)
+                    {
+                        mainListView.Items.Clear();
+                        foreach(TreeNode childNode in node.Nodes)
+                        {
+                            AddNodeToListView(childNode);
+                        }
+                    }
+
+                    if (ListViewItemDoubleClick != null)
+                    {
+                        ListViewItemDoubleClick(sender, new ListViewItemClickEventArgs(info.Item.Tag));
+                    }
+                }
+                else
+                {
+                    mainListView.SelectedItems.Clear();
+                }
+            }
+        }
+
+        /// <summary>
+        /// When the form loads, create the columns that we want to see such as name, file size, and file type.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void TreeViewWithSearchResults_Load(object sender, EventArgs e)
+        {
+            mainListView.Columns.Add("Name");
+            mainListView.Columns.Add("Size");
+            mainListView.Columns.Add("Type");
+            mainListView.SmallImageList = imageList;
+        }
+
+        private void AddNodeToListView(TreeNode node)
+        {
+            ListViewItem item = new ListViewItem(node.Text);
+            item.ImageKey = node.ImageKey;
+            item.Tag = node;
+
+            if (node.Tag.GetType() == typeof(PackageEntry))
+            {
+                var file = node.Tag as PackageEntry;
+                item.SubItems.Add(file.Length.ToFileSizeString());
```

Comment:
@xPaw What do you think about adding a property like `FullLength` or `TotalLength` or something similar which returns your suggestion from `PackageEntry`?